src/PhpStorage.php

```diff
@@ -0,0 +1,657 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Yiisoft\Rbac;
+
+use InvalidArgumentException;
+use Yiisoft\VarDumper\VarDumper;
+
+/**
+ * Class PhpStorage
+ * @package Yiisoft\Rbac
+ *
+ * The authorization data will be saved to and loaded from three files
+ * specified by {@see PhpStorage::itemFile}, {@see PhpStorage::assignmentFile} and {@see PhpStorage::ruleFile}.
+ *
+ * Manager is mainly suitable for authorization data that is not too big
+ * (for example, the authorization data for a personal blog system).
+ */
+final class PhpStorage implements Storage
+{
+    /**
+     * @var string the path of the PHP script that contains the authorization items.
+     * This can be either a file path or a [path alias](guide:concept-aliases) to the file.
+     * Make sure this file is writable by the Web server process if the authorization needs to be changed
+     * online.
+     *
+     * @see loadFromFile()
+     * @see saveToFile()
+     */
+    private string $itemFile;
+    /**
+     * @var string the path of the PHP script that contains the authorization assignments.
+     * This can be either a file path or a [path alias](guide:concept-aliases) to the file.
+     * Make sure this file is writable by the Web server process if the authorization needs to be changed
+     * online.
+     *
+     * @see loadFromFile()
+     * @see saveToFile()
+     */
+    private string $assignmentFile;
+    /**
+     * @var string the path of the PHP script that contains the authorization rules.
+     * This can be either a file path or a [path alias](guide:concept-aliases) to the file.
+     * Make sure this file is writable by the Web server process if the authorization needs to be changed
+     * online.
+     *
+     * @see loadFromFile()
+     * @see saveToFile()
+     */
+    private string $ruleFile;
+
+    /**
+     * @var Item[]
+     * format [itemName => item]
+     */
+    private array $items = [];
+
+    /**
+     * @var array
+     * format [itemName => [childName => child]]
+     */
+    private array $children = [];
+
+    /**
+     * @var array
+     * format [userId => [itemName => assignment]]
+     */
+    private array $assignments = [];
+
+    /**
+     * @var Rule[]
+     * format [ruleName => rule]
+     */
+    private array $rules = [];
+
+    public function __construct(
+        string $directory,
+        string $itemFile = 'items.php',
+        string $assignmentFile = 'assignments.php',
+        string $ruleFile = 'rules.php'
+    ) {
+        $this->itemFile = $directory . DIRECTORY_SEPARATOR . $itemFile;
+        $this->assignmentFile = $directory . DIRECTORY_SEPARATOR . $assignmentFile;
+        $this->ruleFile = $directory . DIRECTORY_SEPARATOR . $ruleFile;
+        $this->load();
+    }
+
+    /**
+     * @return Item[]
+     */
+    public function getItems(): array
```

Comment:
All method phpdoc from here should be moved up to interface. There should be no need to have it in this class.

Reply:
fixed